### PR TITLE
modalai_esc: maintenance release 1

### PR DESCRIPTION
**Pull request by modaltb**
Was having git hell with trying to get things clean.... just made a new PR:

- updated turtle mode to better reflect user experience expectations
- abstract default UART ESC port to boards file
- fix bug with feedback parsing

### CODE REVIEW

**Review by dagar — DISMISSED** [review]

File: src/drivers/actuators/modalai_esc/modalai_esc.cpp [hunk 1]
```diff
@@ -878,167 +882,218 @@ void ModalaiEsc::updateLeds(vehicle_control_mode_s mode, led_control_s control)
...
+	bool use_yaw   = true;
+	bool isolate   = false;
+
+	const float flipPowerFactor = 1.0f - ((float)_parameters.turtle_motor_expo / 100.0f);
+
+	const float stickDeflectionPitchAbs = fabsf(_manual_control_setpoint.x);
+	const float stickDeflectionRollAbs  = fabsf(_manual_control_setpoint.y);
+	const float stickDeflectionYawAbs   = fabsf(_manual_control_setpoint.r);
+
+	const float stickDeflectionPitchExpo = flipPowerFactor * stickDeflectionPitchAbs + (float)pow(stickDeflectionPitchAbs,
```
Comment (dagar) [inline src/drivers/actuators/modalai_esc/modalai_esc.cpp:904]:
```suggestion
	const float stickDeflectionPitchExpo = flipPowerFactor * stickDeflectionPitchAbs + (float)powf(stickDeflectionPitchAbs,
```

File: src/drivers/actuators/modalai_esc/modalai_esc.cpp [hunk 2]
```diff
@@ -878,167 +882,218 @@ void ModalaiEsc::updateLeds(vehicle_control_mode_s mode, led_control_s control)
...
+					       3.0) * (1 - flipPowerFactor);
+	const float stickDeflectionRollExpo  = flipPowerFactor * stickDeflectionRollAbs + (float)pow(stickDeflectionRollAbs,
```
Comment (dagar) [inline src/drivers/actuators/modalai_esc/modalai_esc.cpp:906]:
```suggestion
	const float stickDeflectionRollExpo  = flipPowerFactor * stickDeflectionRollAbs + powf(stickDeflectionRollAbs,
```

File: src/drivers/actuators/modalai_esc/modalai_esc.cpp [hunk 3]
```diff
@@ -878,167 +882,218 @@ void ModalaiEsc::updateLeds(vehicle_control_mode_s mode, led_control_s control)
...
+					       3.0) * (1 - flipPowerFactor);
+	const float stickDeflectionYawExpo  = flipPowerFactor * stickDeflectionYawAbs + (float)pow(stickDeflectionYawAbs,
```
Comment (dagar) [inline src/drivers/actuators/modalai_esc/modalai_esc.cpp:908]:
```suggestion
	const float stickDeflectionYawExpo  = flipPowerFactor * stickDeflectionYawAbs + powf(stickDeflectionYawAbs,
```

**Comment by modaltb** [comment]
Fixed camel cases that were left over, as well as updated some variable names to get a little more separation but providing readability when comparing to other version (goal was to not confuse people)

**Comment by dagar** [comment]
One trivial little CI failure. https://github.com/PX4/PX4-Autopilot/actions/runs/3065032902/jobs/4948726275

Screenshot from 2022-09-16 11-17-31

**Comment by modaltb** [comment]
OK, had the pre-commit astyle installed but it didn't get a style failure apparently.... one more try!!!

**Comment by modaltb** [comment]
Aight, now removed the link to beta flight

**Review by dagar — APPROVED** [review]

**Pull request merged**